src/state/apis/limit-orders/limitOrderApi.ts

```diff
@@ -1,47 +1,152 @@
+import {
+  CoWSwapOrderKind,
+  CowSwapQuoteErrorType,
+  CoWSwapSellTokenSource,
+  CoWSwapSigningScheme,
+  getCowswapNetwork,
+  TradeQuoteError,
+} from '@shapeshiftoss/swapper'
+import {
+  getAffiliateAppDataFragmentByChainId,
+  getFullAppData,
+} from '@shapeshiftoss/swapper/dist/swappers/CowSwapper/utils/helpers/helpers'
+import type { AxiosError } from 'axios'
 import axios from 'axios'
+import { getConfig } from 'config'
+import type { Address } from 'viem'
+import { zeroAddress } from 'viem'
 
 import { BASE_RTK_CREATE_API_CONFIG } from '../const'
-import type {
-  CancelLimitOrdersRequest,
-  CompetitionOrderStatus,
-  GetOrdersRequest,
-  LimitOrder,
-  LimitOrderRequest,
-  Order,
-  Trade,
+import {
+  type CancelLimitOrdersRequest,
+  type CompetitionOrderStatus,
+  type GetOrdersRequest,
+  type LimitOrder,
+  type LimitOrderId,
+  type LimitOrderQuoteRequest,
+  type Order,
+  PriceQuality,
+  type Trade,
 } from './types'
 
+export type LimitOrderQuoteParams = {
+  sellAssetId: AssetId
+  buyAssetId: AssetId
+  chainId: ChainId
+  slippageTolerancePercentageDecimal: string
+  affiliateBps: string
+  sellAccountAddress: Address | undefined
+  sellAmountCryptoBaseUnit: string
+}
+
 export const limitOrderApi = createApi({
   ...BASE_RTK_CREATE_API_CONFIG,
   reducerPath: 'limitOrderApi',
   keepUnusedDataFor: Number.MAX_SAFE_INTEGER, // never clear, we will manage this
   tagTypes: ['LimitOrder'],
   endpoints: build => ({
-    getLimitOrder: build.mutation<
-      LimitOrder,
-      { payload: LimitOrderRequest; config: SwapperConfig }
-    >({
-      queryFn: async ({ payload, config }) => {
+    quoteLimitOrder: build.query<LimitOrder, LimitOrderQuoteParams>({
+      queryFn: async ({
+        sellAssetId,
+        buyAssetId,
+        chainId,
+        slippageTolerancePercentageDecimal,
+        affiliateBps,
+        sellAccountAddress,
+        sellAmountCryptoBaseUnit,
+      }: LimitOrderQuoteParams) => {
+        const config = getConfig()
         const baseUrl = config.REACT_APP_COWSWAP_BASE_URL
-        const maybeNetwork = getCowswapNetwork(payload.chainId)
+        const maybeNetwork = getCowswapNetwork(chainId)
         if (maybeNetwork.isErr()) throw maybeNetwork.unwrapErr()
         const network = maybeNetwork.unwrap()
-        const result = await axios.post<LimitOrder>(`${baseUrl}/${network}/api/v1/orders/`, payload)
+
+        const affiliateAppDataFragment = getAffiliateAppDataFragmentByChainId({
+          affiliateBps,
+          chainId,
+        })
+
+        const { appData, appDataHash } = await getFullAppData(
+          slippageTolerancePercentageDecimal,
+          affiliateAppDataFragment,
+          'limit',
+        )
+
+        const limitOrderQuoteRequest: LimitOrderQuoteRequest = {
+          sellToken: fromAssetId(sellAssetId).assetReference,
+          buyToken: fromAssetId(buyAssetId).assetReference,
+          receiver: undefined, // TODO: implement useReceiveAddress
```

Comment:
This seems a little dangerous. If we forget to fix this TODO might we send funds to a black hole?

Reply:
It's probably fine, there are a bunch of other TODOs in this file.